sstables/sstables.cc

```diff
@@ -2616,10 +2622,14 @@ future<bool> validate_checksums(shared_sstable sst, reader_permit permit) {
             }
         } else {
             auto checksum = co_await sst->read_checksum();
-            if (sst->get_version() >= sstable_version_types::mc) {
-                valid = co_await do_validate_uncompressed<crc32_utils>(data_stream, checksum, digest);
+            if (!checksum) {
+                // If `checksum` is null, it means there is no CRC component in the TOC.
+                // Validation should pass in this case.
+                sstlog.warn("No checksums available for SSTable: {}", sst->get_filename());
```

Comment:
After thinking about this some more, I think we should change the return type of this method from `bool` to `enum class validate_checksum_result { invalid = 0, valid = 1, no_checksum = 3 };`.
It is just cleaner -- there are no surprises.